src/clients/inst_kickoff.rb

```diff
@@ -136,13 +136,18 @@ def main
     def copy_zypp_cache
       return unless Stage.initial
 
+      cache_path = Pkg.ZConfig()["repo_cache_path"] || "/var/cache/zypp"
+      log.info("Zypp cache size: #{`du -h -s #{cache_path.shellescape}`}")
+
+      # Delete the destination cache for consistency and to avoid that the copy
+      # below modifies opened files in place, leading to corruption.
+      ::FileUtils.rm_rf(File.join(Installation.destdir, cache_path))
```

Comment:
Shouldn't `SourceCacheCopyTo` take care of this?

Reply:
It could just pass `--remove-destination` to `cp`, yes.

If you prefer that instead and this PR is still useful despite the latest comments in the bug report, I'll change it